### PR TITLE
Changes required for Beet client PR181

### PR DESCRIPTION
**Pull request by grctest**
The Beet wallet is undergoing a significant uplift: https://github.com/bitshares/beet/pull/181

The changes to the Beet wallet required changes to this library, such as:
* Removing index storage - library users should store this data with their own method eg local storage libraries
  * Still used in the Beet wallet itself, it's just not imposed on the NFT.next.js implementation
* Replacing hand crafted websocket implementation with socket.io for reduced maintenance burden/complexity
* Removed webpack, uses esm rather than commonjs
* Refactor of code structure, now exporting connect & link functions, which after running export the active BeetConnection class to the dev for direct blockchain interaction.
* Using `@noble/ed25519` instead of `eliptic` for ECDH key agreement (primarily due to eliptic's dev inactivity)

To see this in action check out the following example implementation:
https://github.com/BTS-CM/NFT.next.js/blob/main/pages/beet.js
https://github.com/BTS-CM/NFT.next.js/tree/main/components/beet

### CODE REVIEW

**Comment by grctest** [comment]
The [BTS JS injection example](https://github.com/BTS-CM/beet-esm/blob/master/examples/bitshares/inject.js) when implemented [as a web component](https://github.com/BTS-CM/NFT.next.js/blob/main/components/beet/bid.js) returns the following error:

```
{
  "code": 10,
  "message": "Execution error: Assert Exception: now <= trx.expiration: ",
  "data": {
    "code": 10,
    "name": "assert_exception",
    "message": "Assert Exception",
    "stack": [
      {
        "context": {
          "level": "error",
          "file": "db_block.cpp",
          "line": 735,
          "method": "_apply_transaction",
          "hostname": "",
          "thread_name": "th_a",
          "timestamp": "2022-05-18T17:00:00"
        },
        "format": "now <= trx.expiration: ",
        "data": {
          "now": "2022-05-18T17:00:00",
          "trx.exp": "1970-01-01T00:00:00"
        }
      },
      {
        "context": {
          "level": "warn",
          "file": "db_block.cpp",
          "line": 770,
          "method": "_apply_transaction",
          "hostname": "",
          "thread_name": "th_a",
          "timestamp": "2022-05-18T17:00:00"
        },
        "format": "",
        "data": {
          "trx": {
            "ref_block_num": 49362,
            "ref_block_prefix": 3230940728,
            "expiration": "1970-01-01T00:00:00",
            "operations": [
              [
                1,
                {
                  "fee": {
                    "amount": 48260,
                    "asset_id": "1.3.0"
                  },
                  "seller": "1.2.1808745",
                  "amount_to_sell": {
                    "amount": 1,
                    "asset_id": "1.3.0"
                  },
                  "min_to_receive": {
                    "amount": 1,
                    "asset_id": "1.3.6073"
                  },
                  "expiration": "2022-06-18T16:59:54",
                  "fill_or_kill": false,
                  "extensions": []
                }
              ]
            ],
            "extensions": [],
            "signatures": [
              "signature"
            ]
          }
        }
      },
      {
        "context": {
          "level": "warn",
          "file": "db_block.cpp",
          "line": 278,
          "method": "push_transaction",
          "hostname": "",
          "thread_name": "th_a",
          "timestamp": "2022-05-18T17:00:00"
        },
        "format": "",
        "data": {
          "trx": {
            "ref_block_num": 49362,
            "ref_block_prefix": 3230940728,
            "expiration": "1970-01-01T00:00:00",
            "operations": [
              [
                1,
                {
                  "fee": {
                    "amount": 48260,
                    "asset_id": "1.3.0"
                  },
                  "seller": "1.2.1808745",
                  "amount_to_sell": {
                    "amount": 1,
                    "asset_id": "1.3.0"
                  },
                  "min_to_receive": {
                    "amount": 1,
                    "asset_id": "1.3.6073"
                  },
                  "expiration": "2022-06-18T16:59:54",
                  "fill_or_kill": false,
                  "extensions": []
                }
              ]
            ],
            "extensions": [],
            "signatures": [
              "signature"
            ]
          }
        }
      }
    ]
  },
  "digest": "digest",
  "transaction": "transaction"
}
```

The parent TRX has 1970 as expiration, but the operation itself has the correctly formatted expiration. Should a second expiration to overwrite the 1970 value be included? Or is this a bitsharesjs/bitshares-core issue?

**Comment by abitmore** [comment]
> The [BTS JS injection example](https://github.com/BTS-CM/beet-esm/blob/master/examples/bitshares/inject.js) when implemented [as a web component](https://github.com/BTS-CM/NFT.next.js/blob/main/components/beet/bid.js) returns the following error:
> 
> ```
> {
>   "code": 10,
>   "message": "Execution error: Assert Exception: now <= trx.expiration: ",
> }
> ```
> 
> The parent TRX has 1970 as expiration, but the operation itself has the correctly formatted expiration. Should a second expiration to overwrite the 1970 value be included? Or is this a bitsharesjs/bitshares-core issue?

It should have been overwritten already: https://github.com/bitshares/bitsharesjs/blob/94ae9df594897becc1defaa119ed03a1c726dec6/lib/chain/src/TransactionBuilder.js#L153-L156
```
                            if (this.expiration === 0)
                                this.expiration =
                                    base_expiration_sec() +
                                    ChainConfig.expire_in_secs;
```

Perhaps call `set_expire_seconds(sec)` explicitly? See code
https://github.com/bitshares/bitsharesjs/blob/94ae9df594897becc1defaa119ed03a1c726dec6/lib/chain/src/TransactionBuilder.js#L311-L317
```
    /** optional: there is a deafult expiration */
    set_expire_seconds(sec) {
        if (this.tr_buffer) {
            throw new Error("already finalized");
        }
        return (this.expiration = base_expiration_sec() + sec);
    }
```

**Comment by sschiessl-bcp** [comment]
How are you creating a browser compatible bundle to include in plain html/javascript?

**Comment by grctest** [comment]
I've reintroduced the webpack compiled bundle for plain javascript use

**Comment by grctest** [comment]
Reintroduced HTTPS and added a new nft_object signing function and example

File: src/index.js [hunk 1]
```diff
@@ -1,128 +1,156 @@
...
-        } else {
-            let appInstance = new BeetApp(appName);
-            await appInstance.init();
-            this._beetAppInstances[appName] = appInstance;
-            return this._beetAppInstances[appName];
-        }
+    let authToken;
+    try {
+      authToken = await beetConnection.connect(
+        'BTS',
```
Comment (sschiessl-bcp) [inline src/index.js:68]: BTS hardcoded?

File: src/index.js [hunk 2]
```diff
@@ -1,128 +1,156 @@
...
-                returnValue[chain] = await _beetConnectedApp.getAnyConnection(!forceToChoose);
-            } else {
-                returnValue[chain] = await _beetConnectedApp.getChainConnection(chain, !forceToChoose);
-            }
-        }
-        return returnValue;
 
+    if (!authToken) { // fallback to http
+      try {
+        authToken = await beetConnection.connect('BTS', false, httpPort)
```
Comment (sschiessl-bcp) [inline src/index.js:78]: BTS hardcoded?

**Comment by sschiessl-bcp** [comment]
Was there a reason why you not only upgraded dependencies but also refactored pretty much everything? :) It's breaking my head trying to review change, I pretty much need to review the new code as its own..

File: src/index.js (hunk 2 above)
Comment (grctest) [reply src/index.js]: Whoops, replaced with identity variable

**Comment by grctest** [comment]
The move to ESM and Vue3 is to blame for the big refactor, lol.

File: src/index.js (hunk 1 above)
Comment (grctest) [reply src/index.js]: Fixed 👍

**Review by sschiessl-bcp — APPROVED** [review]

**Pull request merged**